File: magma-platform/magma-platform-bungee/src/main/java/de/jalumu/magma/platform/bungee/module/BungeeModuleLoader.java
package de.jalumu.magma.platform.bungee.module;

import de.jalumu.magma.platform.MagmaPlatform;
import de.jalumu.magma.platform.base.module.ModuleLoader;
import de.jalumu.magma.platform.bungee.bootstrap.MagmaBungeeBootstrap;

import java.io.File;

public class BungeeModuleLoader extends ModuleLoader {

    public BungeeModuleLoader(MagmaPlatform platform, File moduleDirectory) {
        super(platform, moduleDirectory);
    }

    @Override
    protected boolean isPlatformPluginAvailable(String plugin) {
        return ((MagmaBungeeBootstrap) getPlatform().getMagmaPluginInstance()).getProxy().getPluginManager().getPlugin(plugin) != null;
    }
}

File: magma-platform/magma-platform-bungee/src/main/kotlin/de/jalumu/magma/platform/bungee/module/BungeeModuleLoader.kt
package de.jalumu.magma.platform.bungee.module

import de.jalumu.magma.platform.MagmaPlatform
import de.jalumu.magma.platform.MagmaPlatform.Companion.platform
import de.jalumu.magma.platform.base.module.ModuleLoader
import de.jalumu.magma.platform.bungee.bootstrap.MagmaBungeeBootstrap
import java.io.File

class BungeeModuleLoader(platform: MagmaPlatform, moduleDirectory: File) : ModuleLoader(platform, moduleDirectory) {
    override fun isPlatformPluginAvailable(plugin: String?): Boolean {
        return (platform?.magmaPluginInstance as MagmaBungeeBootstrap).proxy.pluginManager.getPlugin(plugin) != null
    }
}
